kernel/packages/shared/session/index.ts

```diff
@@ -1,16 +1,56 @@
 import { setLoadingScreenVisible } from 'unity-interface/dcl'
 
 import { disconnect, sendToMordor } from 'shared/comms'
-import { removeUserProfile } from 'shared/comms/peers'
 import { bringDownClientAndShowError } from 'shared/loading/ReportFatalError'
 import { NEW_LOGIN } from 'shared/loading/types'
 import { StoreContainer, RootState } from 'shared/store/rootTypes'
 
 import { getCurrentIdentity, hasWallet as hasWalletSelector } from './selectors'
 import { Store } from 'redux'
+import { getFromLocalStorage, removeFromLocalStorage, saveToLocalStorage } from 'atomicHelpers/localStorage'
+import { StoredSession } from './types'
 
 declare const globalThis: StoreContainer
 
+const SESSION_KEY_PREFIX = 'dcl-session'
+const LAST_SESSION_KEY = 'dcl-last-session-id'
+
+function sessionKey(userId: string) {
+  return `${SESSION_KEY_PREFIX}-${userId}`
+}
+
+export const setStoredSession: (session: StoredSession) => void = (session) => {
+  saveToLocalStorage(LAST_SESSION_KEY, session.userId)
+  saveToLocalStorage(sessionKey(session.userId), session)
+}
+
+export const getStoredSession: (userId: string) => StoredSession | null = (userId) => {
+  const existingSession: StoredSession | null = getFromLocalStorage(sessionKey(userId))
+
+  if (existingSession) {
+    return existingSession
+  } else {
+    // If not existing session was found, we check the old session storage
+    const oldSession: StoredSession | null = getFromLocalStorage('dcl-profile') || {}
+    if (oldSession && oldSession.userId === userId) {
+      setStoredSession(oldSession)
+      return oldSession
+    }
+  }
+
+  return null
+}
+
+export const removeStoredSession = () => removeFromLocalStorage('dcl-profile')
```

Comment:
Would it make sense to extract 'dcl-profile'? Like with `SESSION_KEY_PREFIX`

Reply:
I think this is not used anymore